### PR TITLE
fix(getAnchoredPosition): add better support for floatingElements inside dialogs

### PR DESCRIPTION
**Pull request by francinelucca**
Relates to https://github.com/github/primer/issues/6361

This pull request introduces a new feature to the anchored positioning system, allowing floating elements to be constrained within the visible viewport, which is especially useful for overlays inside dialogs or containers that prevent scrolling. It also adds a comprehensive Storybook demo to showcase this behavior in realistic scenarios.

**Anchored positioning enhancements:**

* Added a new `displayInViewport` option to the `PositionSettings` interface, allowing floating elements to be positioned entirely within the visible browser viewport (`window.innerWidth`/`window.innerHeight`) without requiring scrolling. (`src/anchored-position.ts`, [src/anchored-position.tsR90-R97](diffhunk://#diff-f2513f6730ef68c53ab7e895a329d896afdef2cd6ceeda356633ebe1c82eb6c4R90-R97))
* Updated the core positioning logic in `pureCalculateAnchoredPosition` to use the visible viewport for constraints when `displayInViewport` is true, including left/right and top/bottom nudging and alternate position flipping. (`src/anchored-position.ts`, [[1]](diffhunk://#diff-f2513f6730ef68c53ab7e895a329d896afdef2cd6ceeda356633ebe1c82eb6c4L333-R362) [[2]](diffhunk://#diff-f2513f6730ef68c53ab7e895a329d896afdef2cd6ceeda356633ebe1c82eb6c4L403-R432)
* Set a default value for `displayInViewport` in `positionDefaults` and ensured it is included in `getDefaultSettings`. (`src/anchored-position.ts`, [[1]](diffhunk://#diff-f2513f6730ef68c53ab7e895a329d896afdef2cd6ceeda356633ebe1c82eb6c4R303) [[2]](diffhunk://#diff-f2513f6730ef68c53ab7e895a329d896afdef2cd6ceeda356633ebe1c82eb6c4R323)

**Testing improvements:**

* Added comprehensive test cases to verify `getAnchoredPosition` behavior with `displayInViewport`, including edge cases such as anchors near viewport edges, flipping, inside positioning, and anchors outside the viewport. (`src/__tests__/anchored-position.test.ts`, [src/__tests__/anchored-position.test.tsR576-R699](diffhunk://#diff-966bd08d7c9fc864ddddfa7cd59d9edfa2440704b1985bf2825253e5a1b5827bR576-R699))

**Storybook/demo enhancements:**

* Added a new interactive Storybook story (`InsideDialog`) that demonstrates anchored overlays within a scroll-locked dialog, including realistic content and overlay behavior, along with supporting CSS styles for dialog and overlay presentation. (`src/stories/anchored-position.stories.tsx`, [[1]](diffhunk://#diff-07aeebcefa1e388328bfcd24b8b50232ae89dfaaf14fa6ed29c50153e2b9f969R264-R426); `src/stories/anchored-position.stories.module.css`, [[2]](diffhunk://#diff-92a9430166bc43bf7ba7e33f0a3aea5c62b8a1a951918e9a9551a5b7ae790846R86-R325)

### Testing
- checkout to this PR locally
- `npm run storybook`
- navigate to "Inside Dialog" story
- toggle between "displayInViewport" `true` and `false` in storybook code, notice behavioral difference

### With `displayInViewport=false` (default)

https://github.com/user-attachments/assets/81e21482-3349-4563-aa0e-bcf1a6785bea

### With `displayInViewport=true`

https://github.com/user-attachments/assets/4a2c042a-e25c-402e-addf-07d020f068c9

### CODE REVIEW

File: src/anchored-position.ts [hunk 1]
```diff
@@ -330,14 +340,26 @@ function pureCalculateAnchoredPosition(
   relativePosition: Position,
   floatingRect: Size,
   anchorRect: BoxPosition,
-  {side, align, allowOutOfBounds, anchorOffset, alignmentOffset}: PositionSettings,
+  {side, align, allowOutOfBounds, anchorOffset, alignmentOffset, displayInVisibleViewport}: PositionSettings,
 ): AnchorPosition {
   // Compute the relative viewport rect, to bring it into the same coordinate space as `pos`
```
[the remark this answers is not in the document]
Comment (francinelucca) [reply src/anchored-position.ts]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/primer/behaviors/pull/681#discussion_r2752371210).
